players/g6_player/__init__.py

```diff
@@ -17,187 +18,200 @@ def __init__(
         radius: int,
     ) -> None:
         """Initialise the player with the basic amoeba information
-        Args:
-            maximum_door_frequency (int): the maximum frequency of doors
-            radius (int): the radius of the drone
-        """
 
+            Args:
+                rng (np.random.Generator): numpy random number generator, use this for same player behavior across run
+                logger (logging.Logger): logger use this like logger.info("message")
+                maximum_door_frequency (int): the maximum frequency of doors
+                radius (int): the radius of the drone
+                precomp_dir (str): Directory path to store/load pre-computation
+        """
         self.rng = rng
         self.logger = logger
         self.maximum_door_frequency = maximum_door_frequency
         self.radius = radius
-        self.known_target = False
-        # Data structure to hold state information about the doors inside the radius
-        self.curr_maze = {}
-        self.seen = np.zeros((100, 100))
-        self.curr_pos = (0, 0)
         self.turn = 0
```

Comment:
don't think player has to be concerned about which turn it is - only the maze and doors need that information.